jib-core/src/main/java/com/google/cloud/tools/jib/image/LayerEntry.java

```diff
@@ -36,6 +40,7 @@
 
   private final Path sourceFile;
   private final AbsoluteUnixPath extractionPath;
+  @Nullable private final ImmutableSet<PosixFilePermission> permissions;
```

Comment:
Record this as the integer file mode permissions — it's more compact, and saves the recreation of immutable sets, etc.

Reply:
If we store this as an integer I think we lose the ability to differentiate between using default and explicit permissions without making additional changes (unless we use a nullable `Integer`, which seems weird?)

Reply:
We should be cautious with `-1`, because all the bits are set in the 2's complement binary notation, and it may well be possible that `TarArchiveEntry` only does bit manipulation.

Reply:
-1 indicates that it uses the defaults.   We're not storing TarEntry's special bits here.